llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1127,6 +1134,11 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
   if (EnableSyntheticCounts && !PGOOpt)
     MPM.addPass(SyntheticCountsPropagation());
 
+  if (EnableMarkColdFunctions && PGOOpt &&
+      (PGOOpt->Action == PGOOptions::SampleUse ||
+       PGOOpt->Action == PGOOptions::IRUse))
```

Comment:
It seems that this pass isn't enabled for CS profiles. Can they benefit from this pass?

Reply:
done

Reply:
actually based on the other comment to also apply to manually marked cold functions, now always run this regardless of `PGOOpt`